FormalisingMathematics2025/Section03functions/Sheet1.lean

```diff
@@ -48,44 +43,34 @@ example : ℕ := 2
 -- # But this isn't maths!
 
 -- The type `Prop` contains `Prop`ositions...
-
 #check 2 + 2 = 4
 #check rexp 1 < π
-
 #check 2 + 2 = 5
 #check Irrational (rexp 1 + π)
 #check myFavouriteNumber = yourFavouriteNumber
-
 def MyDifficultProposition : Prop := ∀ n : ℕ, ∃ p, n ≤ p ∧ Prime p ∧ Prime (p + 2)
 def MyEasyProposition : Prop := ∀ n : ℕ, ∃ p, n ≤ p ∧ Prime p ∧ Prime (p + 2) ∧ Prime (p + 4)
-def MyVeryEasyProposition : Prop := ∀ n : ℕ, ∃ p, n ≤ p
 
 -- Key! If `p : Prop`, an expression of type `p` is a proof of `p`.
-
 example : 2 + 2 = 4 := rfl
 example : 2 + 2 ≠ 5 := by simp
 example : ∀ n : ℕ, 2 ≤ n → ∃ x y z : ℕ, 4 * x * y * z = n * (x * y + x * z + y * z) := sorry
--- Erdős-Strauss conjecture
-
-example (n : ℕ) (hn : 2 ≤ n) :
-  ∃ x y z : ℕ, 4 * x * y * z = n * (x * y + x * z + y * z) := sorry
+-- Erdős-Strauss conjecture. We're not going to prove it now (unless you know how?)
 
 -- # How can we make these expressions?
 
 -- Simple proof terms
 example : True := trivial
 example : 2 = 2 := rfl
 example (a b : ℕ) : a + b = b + a := Nat.add_comm a b
-
 example (a b : ℕ) : a * b = b * a := Nat.mul_comm a b
 
-theorem my_proof : MyVeryEasyProposition := fun n => ⟨n, le_rfl⟩
-
+def MyVeryEasyProposition : Prop := 2 = 2
+def my_proof : MyVeryEasyProposition := rfl
```

Comment:
I simplified the proposition itself to be a much simpler one that we've just seen earlier. We're introducing the most important idea here (proofs are of types of their propositions), and I think having any amount of extra complexity here is detracting from the point. In the original example, the first switch to `theorem` syntax (we've never seen it before), the `fun` syntax (we've never seen it before), the `⟨ ⟩` syntax (we've never seen it before), and `le_rfl` (we don't know what this is) all feel to me like they're detracting from the key point we want to convey.

I think it doesn't hurt to readd this example later below again if we want to but it didn't seem essential to me. The examples already below seem to already tie the story more nicely imo.